Remove Ubuntu-18.04 jobs since github deprecated them

TGWDB opened the pull request:
This is in response to [this announcement](https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/).
<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

TGWDB committed:
Remove Ubuntu-18.04 jobs since github deprecated them

thomasspriggs reviewed (commented):
Looks good so far, we should remove the references to Ubuntu 18 from the work flows documentation - https://github.com/diffblue/cbmc/blob/888a08a9f1a6cbe35d31adb798cece5298a12c9d/.github/workflows/README.md?plain=1#L41
and from the release page text - https://github.com/diffblue/cbmc/blob/888a08a9f1a6cbe35d31adb798cece5298a12c9d/.github/workflows/regular-release.yaml#L62
I think the release text also needs Ubuntu 22.04 adding to it, as I think this was missed when we added the 22.04 package to the releases

TGWDB commented:
> Looks good so far, we should remove the references to Ubuntu 18 from the work flows documentation -
> 
> https://github.com/diffblue/cbmc/blob/888a08a9f1a6cbe35d31adb798cece5298a12c9d/.github/workflows/README.md?plain=1#L41
> 
> 
> and from the release page text -
> https://github.com/diffblue/cbmc/blob/888a08a9f1a6cbe35d31adb798cece5298a12c9d/.github/workflows/regular-release.yaml#L62
> 
> 
> I think the release text also needs Ubuntu 22.04 adding to it, as I think this was missed when we added the 22.04 package to the releases

Fixed both, thanks.

NlightNFotis commented on .github/workflows/regular-release.yaml:
````diff
@@ -54,13 +54,8 @@ jobs:
             brew upgrade cbmc
             ```
 
-            ## Ubuntu
````
This change overshot slightly - we shouldn't delete the headers or the instruction text, just the `dpkg` instructions for Ubuntu 18.
